common/app/model/RelatedContent.scala

```diff
@@ -39,13 +39,17 @@ object RelatedContent {
 
 object StoryPackages {
   def apply(parent: ContentType, response: ItemResponse): RelatedContent = {
-    val storyPackagesContent = response.packages.map { packages =>
-      packages.map { p =>
-        p.articles.map(_.content)
+
+    val storyPackagesContent: Seq[ApiContent] = response.packages.map { packages =>
+      val allContentsPerPackage: Seq[Seq[ApiContent]] = packages.map(_.articles.map(_.content))
+      if(packages.size > 0) { //intermix packages only if more than one
+        allContentsPerPackage
+         .flatMap(_.zipWithIndex) // zip content with its position
+         .sortBy(_._2).map(_._1)  // sort by position and extract content
+         .distinct                // remove duplicates
```

Comment:
is this distinct happening on a list of apicontent?  what does content.equals do? can you distinct-by(id) instead of the whole thing?